### PR TITLE
Pdf to text component

### PR DESCRIPTION
**Pull request by willtai**
# Description

[Reopened from this PR](https://github.com/neo4j/neo4j-genai-python/pull/100). Adds a custom `Component` for converting PDF to text. This `Component` can be used as part of the Knowledge Graph construction pipeline.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [x] Unit tests
- [ ] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] Unit tests have been updated
- [ ] E2E tests have been updated
- [x] Examples have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: src/neo4j_genai/components/pdf_loader.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+def is_default_fs(fs: fsspec.AbstractFileSystem) -> bool:
+    return isinstance(fs, LocalFileSystem) and not fs.auto_mkdir
+
+
+class PdfLoader(DataLoader):
+    @staticmethod
+    def load_file(
+        file: Union[Path, str],
+        fs: Optional[AbstractFileSystem] = None,
```
Comment (stellasia) [inline src/neo4j_genai/components/pdf_loader.py:51]: I'm wondering if this `fs` should be exposed somewhere? In the `__init__` of the `DataLoader` maybe?

File: docs/source/user_guide_pipeline.rst [hunk 2]
```diff
@@ -10,9 +10,10 @@ This page provides information about how to create a pipeline.
 
     Pipelines run asynchronously, see examples below.
 
-******************************
+
+*******************
 Creating Components
-******************************
```
Comment (stellasia) [inline docs/source/user_guide_pipeline.rst:15]: Is there a convention for the line length? I feel like we keep changing it from one PR to another :-)

Comment (willtai) [reply docs/source/user_guide_pipeline.rst:15]: I think the convention from what I've seen is it needs to match the length of the text in the section

File: src/neo4j_genai/components/pdf_loader.py (hunk 1 above)
Comment (willtai) [reply src/neo4j_genai/components/pdf_loader.py]: Moved `fs` into `run()` here https://github.com/neo4j/neo4j-genai-python/pull/103/commits/5bfb59ecbc0696d8fb7fc81fa11dd7bb24cc13f4

**Review by stellasia — APPROVED** [review]

**Pull request merged**